cmd/credentialUtil.go

```diff
@@ -125,6 +126,18 @@ func getBlobCredentialType(ctx context.Context, blobResourceURL string, canBePub
 				isPublicResource = true
 			}
 		} else {
+			// Scenario 2: When resourceURL points to a virtual directory.
+			// Check if the virtual directory is accessible by doing GetProperties on container.
+			// Virtual directory can be accessed/scanned only when its parent container is public.
+			bURLParts := azblob.NewBlobURLParts(*resourceURL)
+			bURLParts.BlobName = ""
+			containerURL := azblob.NewContainerURL(bURLParts.URL(), p)
+			if _, err := containerURL.GetProperties(ctx, azblob.LeaseAccessConditions{}); err == nil {
```

Comment:
This statement is repeated in both the blocks. Scenario1 and scenario2 are the same, and should be merged.  A virtual dir can be public if container is public.

Reply:
Done. I concur that code duplicity can be reduced here. I've extracted out the common part of both if...else flow and for scenario 3 only I'm calling blob.GetProperties().